### PR TITLE
Type hints & misc smaller fixes

### PR DESCRIPTION
**Pull request by dcermak**

### CODE REVIEW

**Comment by Vogtinator** [comment]
IMO it's a bit weird to convert only one of the `namedtuple`s

**Comment by dcermak** [comment]
> IMO it's a bit weird to convert only one of the `namedtuple`s

That's the one I cared about :shrug:

**Comment by Vogtinator** [comment]
```
./osclib/origin.py:4:1: F401 'typing.Generator' imported but unused
./osclib/origin.py:4:1: F401 'typing.Tuple' imported but unused
```

File: osclib/comments.py [hunk 1]
```diff
@@ -36,21 +57,21 @@ def _prepare_url(self, request_id=None, project_name=None,
...
         return url
 
-    def _comment_as_dict(self, comment_element):
+    def _comment_as_dict(self, comment_element: ET.Element) -> Comment:
         """Convert an XML element comment into a dictionary.
         :param comment_element: XML element that store a comment.
         :returns: A Python dictionary object.
         """
-        comment = {
+        when_str = comment_element.get('when')
```
Comment (Vogtinator) [inline osclib/comments.py:65]: Can this ever be None?

Comment (dcermak) [reply osclib/comments.py:65]: No idea, there is zero documentation about the schemas of the comments API.

Comment (Vogtinator) [reply osclib/comments.py:65]: In that case I recommend to remove the handling for `None` again. If the value is somehow missing it should fail fast and not at some point later when there's an unexpected missing timestamp somewhere...

**Comment by dcermak** [comment]
Fabian Vogt ***@***.***> writes:

> @Vogtinator commented on this pull request.
>
>
>
>>          """Convert an XML element comment into a dictionary.
>          :param comment_element: XML element that store a comment.
>          :returns: A Python dictionary object.
>          """
> -        comment = {
> +        when_str = comment_element.get('when')
>
> In that case I recommend to remove the handling for `None` again. If the value is somehow missing it should fail fast and not at some point later when there's an unexpected missing timestamp somewhere...

Done.

**Review by Vogtinator — APPROVED** [review]
LGTM, but let's merge on a less read-only day of the week.

**Review by mimi1vx — APPROVED** [review]

**Comment by dirkmueller** [comment]
@dcermak rebase please

**Comment by dcermak** [comment]
Done

**Comment by fcrozat** [comment]
as mentioned on slack, it broke check_tags_in_requests.py:
```
script-executor] Script written into '/go/pipelines/Run.Changelog.Checker/5497b97e-9b94-4d20-ad65-d1aaee788380.sh'.
check_tags_in_requests.py: option --review-mode: invalid choice: 'accept-onpass' (choose from <ReviewChoices.NORMAL: 'normal'>, <ReviewChoices.NO: 'no'>, <ReviewChoices.ACCEPT: 'accept'>, <ReviewChoices.ACCEPT_ONPASS: 'accept-onpass'>, <ReviewChoices.FALLBACK_ONFAIL: 'fallback-onfail'>, <ReviewChoices.FALLBACK_ALWAYS: 'fallback-always'>)
Try 'check_tags_in_requests.py help' for info.
[script-executor] Script completed with exit code: 1.
```

**Comment by DimStar77** [comment]
For reference: this no longer works with osc 0.182.1 from Leap

**Comment by DimStar77** [comment]
> > For reference: this no longer works with osc 0.182.1 from Leap
> 
> also with #2990?

Tested with today's HEAD aka 82c067dc2c5e8ccdf473d82d687bdfff09f8be3e

**Pull request merged**